File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/2. 2231.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

for result in range(1, N + 1):
  # result 의 각 자리 숫자들의 배열
  numArr = list(map(int, list(str(result))))

  if sum(numArr) + result == N:
    output(f"{result}")
    exit(0)

output('0')

File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/2. 2231.js
// 입력
const inputArr = require('fs').readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]

for (let result = 0; result <= N; result++) {
  const numArr = result.toString().split('').map(Number)

  let temp = result

  numArr.forEach(num => {
    temp += num
  })

  if (temp === N) {
    process.stdout.write(`${result}`)
    return
  }
}

process.stdout.write('0')
